### DISCUSSION TITLE
Not able to install and connect with Kurento on Ubuntu 22.04.4 LTS (jammy)

### DISCUSSION THREAD

**Discussion by ajaylashkari** [Kurento/kurento#81] [Community]
Hello actually we are trying to setup the kurentu on my server machines Ubuntu 22.04.4 LTS (jammy)  I have check no package are available for 22.04 So we are try it with docker with running the following command 

docker pull kurento/kurento-media-server:7.1.0

docker run -d --name kurento --network host \
    kurento/kurento-media-server:7.1.0
    
 According  to the docs https://doc-kurento.readthedocs.io/en/latest/user/installation.html#docker-image 
 
 But not get the good results not connected. So please give me the right way how we can set up it on ubuntu 22.04

**Comment by slabajo** [comment]
Please, can you provide the logs and the precise error you are getting? It should work, and in fact among test environments was precisely yours, that is deploying a docker image for kurento 7.1.0 on ubuntu 22.04.

It is true that we have skipped ubuntu 22.04 and we have gone directly from unbuntu 20.04 in kurento 7.0.0 to ubuntu 24.04 for kurento 7.1.0.  In fact you can get packages in an ubuntu 24.04 machine using the usual installation guidelines. 

As you can see, kurento 7.1.0 docker images are based on ubuntu 24.04

**Comment by ajaylashkari** [comment]
Hello @slabajo  

Thank you for your reply actually we have a server that has following configuration 

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy

And we have to test via docker image So my docker container is up and running. In past we used kurento/kurento-media-server - 6.13.0 
That will work fine with Ubuntu 20.xx 

But now we have followed same for ubuntu 22.04 my docker container up and running, and it looks like the following output 

`CONTAINER ID   IMAGE                                 COMMAND                  CREATED             STATUS                    PORTS                                                                                                                                   NAMES
7cec18f8b914   kurento/kurento-media-server:6.13.0   "/entrypoint.sh"         57 minutes ago      Up 42 minutes (healthy)   0.0.0.0:5000-5050->5000-5050/udp, :::5000-5050->5000-5050/udp, 0.0.0.0:8889->8888/tcp, :::8889->8888/tcp                                kurento`

There is no error log we found !
But while testing this we and  unable to connect please check the attached Image once

![kurento](https://github.com/user-attachments/assets/24b4872b-a04c-452d-a709-b035f3d4056f)

Please let me know how we can configure the same for ubuntu 22.04. And we also check this with updated version of kurento 7.1.0 on ubuntu 22.04. But same result 

please help me out !! Thank you!

**Reply by slabajo** [reply]
Taking just a look to the log from web app we can determine that you are really connecting and interacting with kurento media server. The log that says "Creating webRtcPeer and generating local SDP offer" and in fact the others that follow, shows that the application has been able to connect to kurento media server on port 8888 (well, 8889 in external network according to the docker info you sent). So we can verify that Kurento is being used correctly, and the app is able to connect to it.

The problem happens because the application don't seem to be able to connect media flows, so surely there is a problem with networking configuration. This may not be so simple to guess, If server and browser are on the same local network they should be able to connect directly. Perhaps you should remove the ports redirection on the kurento container and just use the host netwrok (--network host).

But if browser and kurento have no direct visibility on the network (they are not on the same network), and you are using some kind of NAT service (, or if you have some element blocking network connections (some firewall or similar), you may need to setup a TURN server.https://doc-kurento.readthedocs.io/en/latest/user/configuration.html?highlight=turn#stun-turn-server

**Comment by ajaylashkari** [comment]
Yes, We have followed this already thing are working fine with local please have a look if there is any change.

`Setup TURN
https://gabrieltanner.org/blog/turn-server/
sudo apt-get update -y
sudo apt-get install coturn
sudo nano /etc/default/coturn
TURNSERVER_ENABLED=1
 
systemctl start coturn
 
nano /etc/turnserver.conf
listening-ip=0.0.0.0
listening-port=3478
user=test:test123
 
sudo service coturn restart
--------------------------------------------
 
https://doc-kurento.readthedocs.io/en/latest/user/intro.html
Setup Kurento with docker on ubuntu 22
https://techviewleo.com/install-kurento-webrtc-server-on-ubuntu/#:~:text=The%20Media%20server%20here%20is%20the%20Kurento%20Media%20Server.&text=That%20is%20it!,recording%2C%20playback%2C%20and%20transmission.

docker run -d --restart unless-stopped\
    --name kurento-6.13.0 \
    -v /root/eduvacancy-recording-upload/recording:/tmp \
    --network host \
    -p 8888:8888/tcp \
    -p 8433:8433 \
    -e KMS_STUN_IP=207.244234.24342.155 \
    -e KMS_STUN_PORT=3478 \
    -e KMS_TURN_URL=admin:Test123@207.244.234234.155:3478?transport=tcp \
    kurento/kurento-media-server:6.13.0
 
docker exec -it kurento-6.13.0 bash
nano /etc/kurento/modules/kurento/WebRtcEndpoint.conf.ini
 
stunServerAddress=host.docker.internal
stunServerPort=3478
turnURL=admin:PLf5ZKbbAJrFdMoCb2p6@host.docker.internal:3478?transport=tcp
 
` 

Thank you!

**Reply by slabajo** [reply]
perhaps you could check on a tab on the same browser you are running the client with this url: chrome://webrtc-internals This will show all inner information about WebRTC connections and specifically you should find an entry about ICE connectivity, if there is an issue preventing ICE connectivity, some relevant information may appear there

**Reply by ajaylashkari** [reply]
ok sure, I will try! Thank you for your support!

**Reply by ajaylashkari** [reply]
Here are the logs please have a look once 
![logs](https://github.com/user-attachments/assets/692ae8b0-0cb3-4683-a472-b3e18333fd43)

**Reply by slabajo** [reply]
This is only partial logs, and the only point we can verify is that a Kurento pipeline is running. As I commented previously I suppose that the problem is for ICE connectivity not for Kurento, and webrtc-internals diagnostics could be very valuable to diagnose the issue
